addons/sale_stock/models/account_move.py

```diff
@@ -33,11 +33,25 @@ def _get_invoiced_lot_values(self):
         all_invoices_amls = current_invoice_amls.sale_line_ids.invoice_lines.filtered(lambda aml: aml.move_id.state == 'posted').sorted(lambda aml: (aml.date, aml.move_name, aml.id))
         index = all_invoices_amls.ids.index(current_invoice_amls[:1].id) if current_invoice_amls[:1] in all_invoices_amls else 0
         previous_amls = all_invoices_amls[:index]
-
-        previous_qties_invoiced = previous_amls._get_invoiced_qty_per_product()
         invoiced_qties = current_invoice_amls._get_invoiced_qty_per_product()
         invoiced_products = invoiced_qties.keys()
 
+        if self.move_type == 'out_invoice':
+            # filter out the invoices that have been fully refund and re-invoice otherwise, the quantities would be
+            # consumed by the reversed invoice and won't be print on the new draft invoice
+            # TODO: decide if we keep payment state or full_reconcile_id
+            # previous_amls = previous_amls.filtered(lambda aml: aml.move_id.payment_state != 'reversed')
+            previous_amls = previous_amls.filtered(lambda aml: not aml.move_id.reversal_move_id.line_ids.full_reconcile_id)
```

Comment:
@smetl I think it's weird to do this when you can just write the previous line even if it was for display purpose.
Also, it means we are asking the ORM to get the reversal_move_id and it's line to read a string for an information we already have on the original move.
(We only want fully reversed invoice and not partial one)

Reply:
You absolutely need to filter on receivable lines yes. Regarding the `full_reconcile_id`, I don't think it works everytime. Having a full here doesn't mean you have exactly a refund for the whole invoice. Suppose an out_invoice of 2000, partial refund of 500 then partial refund of 1500. You'll get a full even you are not reversing the whole invoice

Reply:
@Jinjiu96 I talked to Laurent and he said that if we only want this wort for fully refund invoice (which we do), then it's ok to use the payment_state.

Could you make the change? (ie.: uncomment the line with payment_state, delete the long one and ofc the TODO)

Reply:
Of course! :pray: